src/common.rs

```diff
@@ -80,13 +80,16 @@ pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::SOCK_CLOEXEC;
 pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::empty();
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
-fn make_addr(domain: Domain, sockaddr: &str) -> Result<UnixAddr> {
+fn make_addr(domain: Domain, sockaddr: &str) -> Result<Box<dyn SockaddrLike>> {
```

Comment:
I think this method could keep unchanged. update return value of method make_addr is enough.

Reply:
good call, updated!